packages/perseus/src/widgets/grapher/grapher-validator.ts

```diff
@@ -20,21 +20,36 @@ function grapherValidator(
     }
 
     // We haven't moved the coords
-    if (state.coords == null) {
+    if (userInput.coords == null) {
         return {
             type: "invalid",
             message: null,
         };
     }
 
     // Get new function handler for grading
-    const grader = functionForType(state.type);
-    const guessCoeffs = grader.getCoefficients(state.coords, state.asymptote);
-    const correctCoeffs = grader.getCoefficients(
-        rubric.correct.coords,
-        // @ts-expect-error - TS(2339) - Property 'asymptote' does not exist on type '{ type: "absolute_value"; coords: [Coord, Coord]; }'
-        rubric.correct.asymptote,
-    );
+    const grader = functionForType(userInput.type);
+    let guessCoeffs: number[];
+    if (userInput.type === "logarithm" || userInput.type === "exponential") {
+        guessCoeffs = grader.getCoefficients(
+            userInput.coords,
+            userInput.asymptote,
+        );
+    } else {
+        guessCoeffs = grader.getCoefficients(userInput.coords);
+    }
+    let correctCoeffs: number[];
+    if (
+        rubric.correct.type === "logarithm" ||
+        rubric.correct.type === "exponential"
+    ) {
```

Comment:
By this point we know the user input and rubric types are the same. Rather than having two if/elses, could we just have one?